structuretoolkit/visualize.py

```diff
@@ -143,8 +144,18 @@ def plot3d(
         raise ValueError("plot method not recognized")
 
 
+def _get_frame(cell):
+    vertices = np.array([[0, 0, 0], [0, 0, 1], [0, 1, 1], [0, 1, 0], [0, 0, 0]])
+    return [
+        f(vertices + v) @ cell
+        for f in [lambda x: x, lambda x: np.roll(x, 1, axis=-1)]
+        for v in [0, [1, 0, 0]]
```

Comment:
```suggestion
    vertices = np.array([[0, 0, 0], [0, 0, 1], [0, 1, 1], [0, 1, 0], [0, 0, 0],
                                    [1, 0, 0], [1, 0, 1], [1, 1, 1], [1, 1, 0], [1, 0, 0] ])
    return [
        f(vertices + v) @ cell
        for f in [lambda x: x, lambda x: np.roll(x, 1, axis=-1)]]
```

I find it difficult to follow this loop here.  Since the loops are not dependent on the function argument, I would simply precompute `vertices` fully and then do `return vertices @ cell`.

Reply:
I also find it difficult to follow

Reply:
I changed it. It's still programmatically determined, and might not be quite straightforward, but I also added tests, so it should be fine.